### PR TITLE
Docs: Document CLI version support

### PR DESCRIPTION
**Pull request by kylegach**
## What I did

Document CLI version support.

#### Manual testing

1. Follow the steps in the [contributing instructions](https://storybook.js.org/docs/react/contribute/new-snippets#preview-your-work) for this branch, `docs-cli-versioning`

### Documentation

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [x] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [x] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Comment by kylegach** [comment]
@JReinhold — Should this be patched? I saw some notes that the functionality works in `7.6`, but wasn't confident in that.

**Review by chantastic — APPROVED** [review]

File: docs/configure/upgrading.md [hunk 1]
```diff
@@ -22,6 +22,12 @@ To help ease the pain of keeping Storybook up-to-date, we provide a command-line
...
 
+The `upgrade` command will use whichever version you specify. For example:
+
+- `npx storybook@latest upgrade` will upgrade to the latest version
+- `npx storybook@7.6.10 upgrade` will upgrade to `7.6.10`
+- `npx storybook@7 upgrade` will upgrade to the newest `7.x.x` version
+
 After running the command the script will:
 
 - Upgrade all Storybook packages in your project to the latest stable version
```
Comment (JReinhold) [inline docs/configure/upgrading.md:33]: This line needs to be corrected/removed. Not sure what a good wording would be though.

**Review by JReinhold — APPROVED** [review]
Looks good! Just one line we need to change, but I'm sure you'll figure it out.

**Comment by JReinhold** [comment]
Yes, this should be patched @kylegach, it's available in v7.6.9 and onwards.

File: docs/api/cli-options.md [hunk 2]
```diff
@@ -112,12 +112,14 @@ Options include:
```
Comment (kylegach) [inline docs/api/cli-options.md:112]: TK: Add similar updates to `init` command, once available.

**Review by jonniebigodes — APPROVED** [review]
Thanks for taking the time to adjust the documentation and add in the versioning changes. Appreciate it 🙏 ! All is good on my end. Merge when able

**Pull request merged**